### ISSUE TITLE
Control character such as Ctrl-^ is not properly reported.

### ISSUE THREAD

**Issue by lambdalisue** [rhysd/NyaoVim#87]
It seems [Control character](https://ja.wikipedia.org/wiki/%E5%88%B6%E5%BE%A1%E6%96%87%E5%AD%97) with non alphabet character is not reported correctly. I use the following minimum vimrc.

```vimrc
" vim -u ~/.vim/vimrc.min
if has('vim_starting')
  set nocompatible
endif

filetype plugin indent on
syntax on

" Check keycode
function! s:getchar() abort
  redraw | echo 'Press any key: '
  let c = getchar()
  while c ==# "\<CursorHold>"
    redraw | echo 'Press any key: '
    let c = getchar()
  endwhile
  redraw | echo printf('Raw: %s | Char: %s', c, nr2char(c))
endfunction
command! GetChar call s:getchar()
```

And

1. Start with `nayovim -u ~/.vimrc.min`
2. Hit `:GetChar`
3. Hit `Ctrl + 6` (`Ctrl-^`)

### Expected Behavior
Shows `Raw 30 | Char ^^`

### Actual Behavior
Shows `<82><fc>^D6 | Char`

### Your Environment

- OS: Mac OS X El capitan
- NyaoVim version: 0.0.22
- `nvim` version: 0.1.6-dev
- Keyboard layout: US

### Additional Comments (if any)
```
NyaoVim version 0.0.22
  electron : 1.4.1
  chrome : 53.0.2785.113
  node : 6.5.0
  v8 : 5.3.332.45
```

`echo "\<C-^>"` shows `^^` in nyaovim and it is different from above so `nmap <C-^> xxxxx` or something like that won't work.

**Comment by rhysd** [comment]
I checked this.

NyaoVim looks to send `<C-^>` properly.

![2016-10-03 9 34 13](https://cloud.githubusercontent.com/assets/823277/19025019/c8d5017a-894c-11e6-9bb9-ab30f4be91b1.png)

Could you check below points?

- Is this can be reproduced on `nvim` in terminal?  If so, it may be a problem of Neovim.
- Please check the sending key sequence in DevTools log with `NODE_ENV=debug nyaovim`.

**Comment by lambdalisue** [comment]
It seems my explanation was not enough sorry 😞 

I also confirm that hitting `Ctrl+Shift+6` send `Ctrl+^` in NyaoVim but I ment that **hitting `Ctrl+6` should send `Ctrl+^` instead of `Ctrl+6`**.

This behavior is confirmed in vim/nvim/GVim/MacVim

http://vim-jp.org/vimdoc-ja/editing.html#CTRL-^

**Note:**
I found that even `Ctrl+6` send `Ctrl+^` even in JIS keyboard (an alternative key of 6 is &) in my environment. If this is true, probably neovim-component should send an fixed alternative key of individual `Ctrl+{digit}` I guess.

**Comment by rhysd** [comment]
> I ment that hitting Ctrl+6 should send Ctrl+^ instead of Ctrl+6.

Is that a Vim feature?  Is it true that Vim frontend should replace ctrl+6 with ctrl+^?  If so, it's very hard because we can't know which layout of keyboard is being used and what the alternate key for ctrl+^.  For example, alternate key for ^ is 6 in US layout, but the one for JP keyboard is ~.

**Comment by rhysd** [comment]
Thank you for additional description.  I think I could understand. 

> I found that even Ctrl+6 send Ctrl+^ even in JIS keyboard (an alternative key of 6 is &) in my environment. If this is true, probably neovim-component should send an fixed alternative key of individual Ctrl+{digit} I guess.

 That's good news.  Let me confirm the behavior before implementing that (is it true that frontend should consider that? etc...).  If neovim-qt is also handling it, it's strong proof of it :)

**Comment by lambdalisue** [comment]
>  If neovim-qt is also handling it, it's strong proof of it :)

I think you should follow GVim/MacVim and Vim's help rather than neovim-qt while neovim-qt is under development and it has [a bug for some mappings](https://github.com/equalsraf/neovim-qt/issues/134)

I think the proof is the following sentence on http://vimdoc.sourceforge.net/htmldoc/editing.html#CTRL-^

> Mostly the ^ character is positioned on the 6 key,
> pressing CTRL and 6 then gets you what we call CTRL-^.

**Comment by rhysd** [comment]
Yeah, the behavior should be implemented anywhere, it's correct.  What I need to know is that it should be done in frontend.  If it should be handled in Vim or Neovim core, it's simply Neovim issue.  I want to clarify it.

**Comment by lambdalisue** [comment]
**Memo**: https://github.com/vim/vim/blob/d58b0f982ad758c59abe47627216a15497e9c3c1/src/gui_w32.c#L1956-L1989

**Comment by rhysd** [comment]
@lambdalisue 

Thank you so much for the investigation.  It's very helpful.  So we need to handle `@`, `^` and `-` in frontend level.  I'll fix them in neovim-component package tonight!

**Comment by lambdalisue** [comment]
@mattn gave me a direction. I would like to express my gratitude :-)

**Comment by rhysd** [comment]
This should be fixed in neovim-component@0.8.1. I'll close this. If the problem is till there, please notify me by creating a new issue or pinging me.
